### ISSUE TITLE
Test and Score 

### ISSUE THREAD

**Issue by arvind-prasad68** [biolab/orange3#3098]
##### Orange version
Orange version 3.13

![image](https://user-images.githubusercontent.com/40472222/41967419-32d1f26c-7a1f-11e8-8e03-d2490337193c.png)
The value of F1 is shown as 0.786 for logistic regression. 

F1 is the harmonic mean, of recall and precision
https://docs.orange.biolab.si/3/data-mining-library/reference/evaluation.cd.html
The value is 2/(1/0.821 + 1/0.808)=0.8144 
The value is incorrect. Request you to please rectify the bug.

**Comment by PrimozGodec** [comment]
I think it is not a bug just weighted averaging is used in computing F1, precision or recall. That means that score is computed per each label/class and then averaged. See http://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html

In case of the classifier with 3 classes, the F1 score is computed for class 1 as a target class using the rule you wrote, class 2 as a target class and class 3 as a target class. Those scores are then averaged to retrieve the final score. Recall and precision are computed same way, all of them are averaging over class scores. The rule you wrote is used in per class calculation but it does not hold for averages. 

If you let me know how to reproduce your results I can give you a more detailed explanation.

**Comment by markotoplak** [comment]
@PrimozGodec, perhaps a more detailed explanation should go into widget documentation? At least the one you wrote here, if it is not of course already there.

**Comment by PrimozGodec** [comment]
@markotoplak Agree with you. I will add that to the documentation.

**Comment by arvind-prasad68** [comment]
Thank @PrimozGodec  and @markotoplak. Your clarifications were of a great help. I have satisfied myself that the values of F1 are in order.

**Comment by PrimozGodec** [comment]
@arvind-prasad68 There is also *Target Class* dropdown in the bottom of the widget. Widget computes scores with averaging until you choose a specific target class. When a target class is selected scores are computed with the respect to the target class. When choosing a target class the formula for the F1 score holds since no averaging is used.

**Comment by ajdapretnar** [comment]
Documentation has been updated.
